File: serverless/rust_app/src/types/testimonial.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub enum Stars {
    Zero,
    Half,
    One,
    OneAndAHalf,
    Two,
    TwoAndAHalf,
    Three,
    ThreeAndAHalf,
    Four,
    FourAndAHalf,
    Five,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct Testimonial {
    #[serde(rename = "PK")]
    pub primary_key: String,
    #[serde(rename = "SK")]
    pub sort_key: String,
    #[serde(rename = "GSI-PK")]
    pub gsi_primary_key: String,
    #[serde(rename = "GSI-SK")]
    pub gsi_sort_key: String,
    pub name: Option<String>, // Backwards compatibility, locally
    pub stars: f32,
    pub title: String,
    pub comment: String,
    pub active: bool,
    pub created: String,
    pub modified: Option<String>,
}

#[derive(Deserialize)]
pub struct TestimonialPutRequest {
    pub name: String,
    pub stars: Stars,
    pub title: String,
    pub comment: String,
    pub active: bool,
}

#[derive(Deserialize)]
pub struct TestimonialToggleRequest {
    #[serde(rename = "PK")]
    pub primary_key: String,
    #[serde(rename = "SK")]
    pub sort_key: String,
    pub active: bool,
}

File: frontend/src/types/testimonial.ts
import { IMandatoryDynamodbFields } from ".";

export enum EStars {
  Zero = "Zero",
  Half = "Half",
  One = "One",
  OneAndAHalf = "OneAndAHalf",
  Two = "Two",
  TwoAndAHalf = "TwoAndAHalf",
  Three = "Three",
  ThreeAndAHalf = "ThreeAndAHalf",
  Four = "Four",
  FourAndAHalf = "FourAndAHalf",
  Five = "Five",
}

export interface ITestimonialPutRequest {
  name: string;
  stars: EStars;
  title: string;
  comment: string;
  active: boolean;
}

export interface ITestimonial extends ITestimonialPutRequest, IMandatoryDynamodbFields { }
